lxd/events.go

```diff
@@ -35,7 +35,31 @@ func (r *eventsServe) String() string {
 }
 
 func eventsSocket(d *Daemon, r *http.Request, w http.ResponseWriter) error {
+	allProjects := queryParam(r, "all-projects") == "true"
+	projectQueryParam := queryParam(r, "project")
+	if allProjects && projectQueryParam != "" {
+		response.BadRequest(fmt.Errorf("'project' cannot be set if 'all-projects' is set"))
+		return nil
+	}
+
+	if !allProjects && projectQueryParam != "" {
+		_, err := d.cluster.GetProject(projectQueryParam)
+		switch err {
+		case nil:
+			break
+		case db.ErrNoSuchObject:
+			response.BadRequest(fmt.Errorf("project '%s' does not exist", projectQueryParam)).Render(w)
+			return nil
+		default:
+			return err
+		}
+	}
+
 	projectName := projectParam(r)
+	if allProjects {
+		projectName = "*"
```

Comment:
Not originally requested, but given how difficult it is to find where "*" is used in the code base in relation to this, could you add a constant that represents this, e.g. `EventsAllProjects` or something like that, and then find where it is checked for and use the constant there too for ease of finding it in the future.

Reply:
Actually looking at the change to `AddListener` I don't think we need this set to `*` anymore and should just be removed.

Reply:
I've changed this a little so that it doesn't call `projectParam` anymore, since that will return the default project if none are specified.